src/components/button/Button.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import styled from "styled-components";
+
+type ButtonPropsType = {
+    type: "button" | "submit" | "reset" | undefined
+    disabled?: boolean
+    onClick?: (e?: any) => void
+    styledComponent: any
+}
+
+const Button: React.FC<ButtonPropsType> = ({
+                                               children,
+                                               type,
+                                               disabled,
+                                               onClick,
+                                               styledComponent
+                                           }) => {
+    const StyledButton = styled(styledComponent)``;
```

Comment:
For what is this wrapper?

Reply:
It is needed to send styled component as a prop. I`ll delete it in task-4